modules/instrument_list/php/instrument_list.class.inc

```diff
@@ -50,18 +50,13 @@ class Instrument_List extends \NDB_Menu_Filter
     /**
     * Checking permissions
     *
+    * @param \User $user The user whose access is being checked
+    *
     * @return bool
     */
-    function _hasAccess()
+    function _hasAccess(\User $user) : bool
     {
-        // create user object
-        $user =& \User::singleton();
-        if (isset($_REQUEST['sessionID'])) {
-            //These variable are only for checking the permissions.
-            $TimePoint =& \TimePoint::singleton($_REQUEST['sessionID']);
-            $candID    = $TimePoint->getCandID();
-            $Candidate =& \Candidate::singleton($candID);
-            return ($user->hasPermission('access_all_profiles')
```

Comment:
this reverses changes from https://github.com/aces/Loris/pull/3721. these variables need to be defined